subsys/nrf_security/src/drivers/cracen/cracenpsa/src/kmu.c

```diff
@@ -607,6 +626,20 @@ psa_status_t cracen_kmu_get_builtin_key(psa_drv_slot_number_t slot_number,
 		kmu_opaque_key_buffer *key = (kmu_opaque_key_buffer *)key_buffer;
 
 		key->key_usage_scheme = metadata.key_usage_scheme;
+		switch (metadata.size) {
+		case METADATA_ALG_KEY_BITS_128:
+			key->number_of_slots = 1;
+			break;
+		case METADATA_ALG_KEY_BITS_192:
+		case METADATA_ALG_KEY_BITS_256:
+			key->number_of_slots = 2;
+			break;
+		case METADATA_ALG_KEY_BITS_384_SEED:
+			key->number_of_slots = 3;
+			break;
+		default:
+			return PSA_ERROR_DATA_INVALID;
+		}
 		key->number_of_slots = metadata.size > METADATA_ALG_KEY_BITS_128 ? 2 : 1;
```

Comment:
I don't get why we set `key->number_of_slots` in the switch case before just to set it here to either two or one.

Reply:
this line should be removed :)